src/lib/is_val_object.js

```diff
@@ -0,0 +1,15 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+// returns true for a valid value object and false for tree nodes in the attribute hierarchy
+module.exports = function(obj) {
+    return obj && obj.valType !== undefined;
```

Comment:
I moved it to `coerce.js` and similarly to how other `coerce` constituents are handled, I directly reexport as `Lib.isValObject`. If it's desirable, I can phase out `PlotSchema.isValObject` in favor of `Lib.isValObject`, what do you think?